src/Solution/DirectorySolution.php

```diff
@@ -62,6 +80,17 @@ public static function fromDirectory($directory, $entryPoint = 'solution.php')
     {
         return new static($directory, $entryPoint);
     }
+
+    /**
+     * @param array  $exclusions
+     * @param string $directory
+     * @param string $entryPoint
+     * @return static
+     */
+    public static function fromDirectoryWithExclusions(array $exclusions, $directory, $entryPoint = 'solution.php')
```

Comment:
I don't think we need this extra factory method. Can we just keep the existing one and update the method?

```php
public static function fromDirectory($directory, array $exclusions = [], $entryPoint = 'solution.php');
```

And also can we add a few defaults. So in the above method we could do:

```php
return new static($directory, $entryPoint, array_marge($exclusions, ['composer.lock', 'vendor']));
```

If the consumer doesn't want the defaults then they can just use the constructor manually